apps/wallet/src/ui/app/pages/swap/GasFeeSection.tsx

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import {
+	Coins,
+	getUSDCurrency,
+	SUI_CONVERSION_RATE,
+	useBalanceConversion,
+} from '_app/hooks/useDeepBook';
+import { Text } from '_app/shared/text';
+import { DescriptionItem } from '_pages/approval-request/transaction-request/DescriptionList';
+import { GAS_TYPE_ARG } from '_redux/slices/sui-objects/Coin';
+import { useCoinMetadata, useFormatCoin } from '@mysten/core';
+import { SUI_TYPE_ARG } from '@mysten/sui.js/utils';
+import BigNumber from 'bignumber.js';
+import { useMemo } from 'react';
+
+import { FEES_PERCENTAGE } from './utils';
+
+export function GasFeeSection({
+	activeCoinType,
+	totalGas,
+	amount,
+	isValid,
+}: {
+	activeCoinType: string | null;
+	amount: string;
+	isValid: boolean;
+	totalGas: string;
+}) {
+	const { data: activeCoinData } = useCoinMetadata(activeCoinType);
+	const isAsk = activeCoinType === SUI_TYPE_ARG;
+
+	const estimatedFees = useMemo(() => {
+		if (!amount || !isValid) {
+			return null;
+		}
+
+		return new BigNumber(amount).times(FEES_PERCENTAGE);
+	}, [amount, isValid]);
+
+	const { rawValue } = useBalanceConversion(
+		estimatedFees,
+		isAsk ? Coins.SUI : Coins.USDC,
```

Comment:
Do we only support sui and usdc?

Reply:
yep, for now, there is no liquidity on any other coins

Reply:
@mamos-mysten yes, eventually, but there is no liquidity on the other tokens and the code `should` support easy changes if it comes to that.